business_requirement_sale/wizard/business_requirement_create_sale.py

```diff
@@ -135,10 +132,31 @@ def _prepare_sale_order_vals(self):
     def _create_sale_order(self):
         self.ensure_one()
         vals = self._prepare_sale_order_vals()
-        for deliverable in self.deliverable_ids:
-            line_vals = self._prepare_sale_order_line_vals(deliverable, vals)
-            vals['order_line'].append((0, 0, line_vals))
+        vals_order_line = {'order_line': []}
         order = self.env['sale.order'].create(vals)
+        deliverable_sections = self.deliverable_ids.filtered('section_id')
+        deliverable_undefined = self.deliverable_ids - deliverable_sections
+        if deliverable_undefined:
+            for deliverable in deliverable_undefined:
+                line_vals = self._prepare_sale_order_line_vals(
+                    deliverable, vals)
+                vals_order_line['order_line'].append((0, 0, line_vals))
+            order.write(vals_order_line)
+        if deliverable_sections:
+            for section in self.section_ids:
+                line_vals = self._prepare_sale_order_line_section_vals(
+                    section, vals)
+                order.write({
+                    'order_line': [(0, 0, line_vals)],
+                })
+                for deliverable in deliverable_sections.filtered(
+                    lambda s: s.section_id.id == section.id
+                ):
+                    line_vals = self._prepare_sale_order_line_vals(
+                        deliverable, vals)
+                    order.write({
```

Comment:
Why are you doing several write on the order? This reduces the performance a lot. And you haven't still set the sequence field I mentioned. You must prepare everything in the same loop, ordering deliverables by section and the rest of criteria (not grouping), and then when you detect a change in the section, calling the prepare section method for adding it before, with the corresponding sequence.